lib/pure/httpclient.nim

```diff
@@ -1261,8 +1268,9 @@ proc downloadFile*(client: HttpClient | AsyncHttpClient,
                    url: string, filename: string): Future[void] {.multisync.} =
   ## Downloads ``url`` and saves it to ``filename``.
   client.getBody = false
-  defer:
-    client.getBody = true
+# disabled because this causes an abort after an async download
```

Comment:
There is a reason for this code. Disabling it will cause other issues.

IIRC the reason for this code is to allow a normal ``get`` request after a ``downloadFile``.

Reply:
yes issue #6284